dashboard/modules/serve/serve_head.py

```diff
@@ -1,30 +1,45 @@
 import logging
+from enum import Enum
 from typing import Optional
 
 import aiohttp
 from aiohttp.web import Request, Response
 
-from ray.dashboard.datacenter import DataOrganizer
 import ray.dashboard.optional_utils as dashboard_optional_utils
 import ray.dashboard.utils as dashboard_utils
+from ray.dashboard.datacenter import DataOrganizer
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
-routes = dashboard_optional_utils.ClassMethodRouteTable
+routes = dashboard_optional_utils.DashboardHeadRouteTable
+
+
+class RestMethod(str, Enum):
+    GET = "GET"
+    PUT = "PUT"
+    DELETE = "DELETE"
 
 
 class ServeHead(dashboard_utils.DashboardHeadModule):
     def __init__(self, dashboard_head):
         super().__init__(dashboard_head)
         self._http_session = aiohttp.ClientSession()
+        self.rest_method_executors = {
+            RestMethod.GET: self._http_session.get,
+            RestMethod.PUT: self._http_session.put,
+            RestMethod.DELETE: self._http_session.delete,
+        }
 
-    @routes.get("/api/serve_head/applications/")
```

Comment:
@zcin @edoakes Heads up– this route is changing to [`/api/serve/applications/`](https://github.com/ray-project/ray/pull/39923/files#diff-f7f7a91d0932a8b3bc1572fbefc8367a2132aee24df29ba1792670b20db06eabR98) to match the route on the dashboard agent. Technically, this is an API change, so I want to flag it.

Let me know if I should keep it as is.

Reply:
Messaged you about this one